guides/predictive-analytics-with-deepnote.md

```diff
@@ -0,0 +1,119 @@
+
+<figure><img src="../.gitbook/assets/.predictive-analytics/Integrating postgres.png" alt=""><figcaption><p>Adding integrations</p></figcaption></figure>
+
+<figure><img src="../.gitbook/assets/.predictive-analytics/connecting postgres.png" alt=""><figcaption><p>Inputing your Hydra credentials</p></figcaption></figure>
+
+5. After establishing the connection, you can view the imported salary table within Deepnote.
+Using the SELECT statement, you can run SQL queries on the salary table, enabling you to retrieve and analyze specific data from the table.
+
+```sql
+SELECT * FROM public.salary LIMIT 100
+```
+
+<figure><img src="../.gitbook/assets/.predictive-analytics/Hydra dashboard.png" alt=""><figcaption><p>Deepnote dashboard after connection</p></figcaption></figure>
+
+6. To access the table using Python scripts, an engine has to be created with the data warehouse credentials
+
+```python
+import pandas as pd
+from sqlalchemy import create_engine
+#user = username
+#pass = password
+#server = hostname
+#database = database-name
+engine = create_engine('postgresql://user:pass@server:5432/database')
+#table_name = Salary
+#database = public
+data = pd.read_sql_table('table_name',con= engine, schema= 'database')
+```
+
+**Congratulations!** You've successfully connected Hydra with Deepnote! What next?
+
+## Making the predictions
+
+Now that the Database is connected we can use Python and SQL scripts to access the data. Using the Linear Regression model, we can predict the salary data.
+The necessary models and libraries are being imported.The linear regression model and matplotlib for our visualization are imported as shown below&#x20;
+
+```python
+#import Classes and lib needed
+import matplotlib.pyplot as plt
+from sklearn.linear_model import LinearRegression
+```
+
+The dataset is split into dependent and independent variables(X, Y),the model is trained on the dataset, and prediction on the salary table  is made.
+
+
+```python
+#defining variables X and Y
+X = data.iloc[:,0].values.reshape(-1,1)
+print(X)
+Y = data.iloc[:,1].values.reshape(-1,1)
+print(Y)
+```
+
+```python
+#creating a Linear Regression model for prediction
+model = LinearRegression()
+
+#Fitting the model on the Dataset
+model.fit(X,Y)
+
+Y_pred = model.predict(X)  # make predictions
+```
+
+Visualizations are made on the actual salary and the predicted to view the difference.
+
+```python
+Y_pred = Y_pred.reshape(-1)
+difference = pd.DataFrame({'Real' : data. salary, 'Predicted': Y_pred, })
+print(difference)
+```
+
+```python
+graph = difference.head(10)
+graph.plot(kind='bar')
+plt.title('Actual vs Predicted')
+plt.ylabel('salary')
+```
+
+<figure><img src="../.gitbook/assets/.predictive-analytics/Output.png" alt="" width="464"><figcaption><p>Visualization</p></figcaption></figure>
+
+Your first prediction is made and you can determine the range of a person's salary from years of experience.
```

Comment:
Your first prediction is complete! You can determine the range of a person's salary based on years of experience.

Section - Conclusion
At this point, we have performed a basic predictive analytics project by integrating the Hydra database with Deepnote. This basic guide can be leveraged to experiment with your own business metrics, but don't hesitate to reach out for more advance support through the Hydra #experts channel in Discord.